### PR TITLE
Add compatibility for Django 1.9

### PR DESCRIPTION
**Pull request by thoas**
- Use the new format for patterns
- Drop support for Django 1.4, useless.

I'm also in favor of dropping support for < 1.8, but not my application ;)

### CODE REVIEW

**Comment by k4nar** [comment]
Thank you for your contribution! 

@thoas : Do this PR really break Django 1.4 support ? Because the tests are still passing.

File: mail_factory/tests/test_mails.py [hunk 1]
```diff
@@ -158,11 +158,9 @@ class TestMail(BaseMail):
 
         test_mail = TestMail()
         attachments = [
-            (finders.find('admin/img/nav-bg.gif'), 'nav-bg.gif', 'image/png'),
             (finders.find('admin/css/base.css'), 'base.css', 'plain/text')]
         msg = test_mail.create_email_msg([], attachments=attachments)
         self.assertEqual(len(msg.attachments), 1)  # base.css
-        self.assertEqual(len(msg.related_attachments), 1)  # nav-bg.gif
```
Comment (jpic) [inline mail_factory/tests/test_mails.py]: No more attachement ? Why is this necessary ?

File: mail_factory/urls.py [hunk 2]
```diff
@@ -24,4 +23,4 @@
     url(r'^html_not_found/(?P<mail_name>.*)/$',
         html_not_found,
         name='mail_factory_html_not_found'),
-)
+]
```
Comment (jpic) [inline mail_factory/urls.py]: That doesn't work with django < 1.8 does it ? I have nothing against it, but in that case shouldn't you bump the requirements in the README ?

File: mail_factory/tests/test_mails.py [hunk 3]
```diff
@@ -158,11 +158,9 @@ class TestMail(BaseMail):
 
         test_mail = TestMail()
         attachments = [
-            (finders.find('admin/img/nav-bg.gif'), 'nav-bg.gif', 'image/png'),
```
Comment (jpic) [inline mail_factory/tests/test_mails.py]: Why is this gone ?
